inc/class-core-sitemaps-provider.php

```diff
@@ -110,9 +110,20 @@ public function get_sitemap_type_data() {
 
 		$object_subtypes = $this->get_object_subtypes();
 
+		// If there are no object subtypes, include a single sitemap for the
+		// entire object type.
+		if ( empty( $object_subtypes ) ) {
+			$sitemap_data[] = array(
+				'name'  => '',
+				'pages' => $this->max_num_pages(),
+			);
+			return $sitemap_data;
+		}
+
+		// Otherwise, include individual sitemaps for every object subtype.
 		foreach ( $object_subtypes as $object_subtype_name => $data ) {
 			$sitemap_data[] = array(
-				'name'   => $object_subtype_name,
+				'name'  => $object_subtype_name,
 				'pages' => $this->max_num_pages( $object_subtype_name ),
```

Comment:
Looks like `$object_subtype_name` needs to be explicitly cast to a string?

```
Parameter #1 $object_subtype of method                              
         Core_Sitemaps_Provider::max_num_pages() expects string, int|string  
         given. 
```

Reply:
Where would this be an integer? This should always be a string, if it's not, we may need to fix something elsewhere.

Reply:
The static analysis (PHPStan) doesn't know that `$object_subtypes` is an associative array keyed with strings.

Not sure why it wasn't flagged before. But `$this->max_num_pages( (string) $object_subtype_name )` should do the trick.

Reply:
Not familiar with that tool, seems a bit odd to me if you can't configure it like that - or it's configured overly strict here, cause this is a false positive. But anyway, let's do the string cast then.